message_edit: Update placement and styling of server message edit banners.

shameondev opened the pull request:
<!-- Describe your pull request here.-->
This PR aims to update the old-style custom error banners in the message edit form to the ones we are currently using. It also fixes the placement of the banners.

Fixes: #25412.

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
<img width="741" alt="image" src="https://user-images.githubusercontent.com/53193850/235827549-55ccf19c-90d2-4187-8596-fea6f4531ed2.png">

timabbott commented:
We can probably cut the "Error saving edit" wrapping text -- that would match what we do in compose.

timabbott commented on web/templates/topic_edit_form.hbs:
```diff
@@ -5,6 +5,5 @@
       autocomplete="off" maxlength="{{ max_topic_length }}" />
     <button type="button" class="topic_edit_save small_square_button animated-purple-button"><i class="fa fa-check" aria-hidden="true"></i></button>
     <button type="button" class="topic_edit_cancel small_square_button small_square_x"><i class="fa fa-remove" aria-hidden="true"></i></button>
-    <div class="alert alert-error edit_error" style="display: none"></div>
```
This is the little edit form for changing the topic in a recipient bar -- did you test this? My guess is these code changes won't work for it.

shameondev replied:
Hmm, I don't see `#compose_banners` in this template, so I agree that it may not work. I need to review it more carefully.

alya replied:
Does this actually come up? I wouldn't worry about changing it if it almost never happens; there is nothing particularly wrong with "before" (whereas "after" is clearly problematic).

alya commented:
> We can probably cut the "Error saving edit" wrapping text -- that would match what we do in compose.

Yes, definitely, we should match the compose banner.

shameondev commented:
I've removed the "Error saving edit" prefix and reverted the changes made to the inline topic edit form, as there are ongoing discussions regarding it.

alya commented:
Looks good to me in manual testing.

timabbott commented on web/src/message_edit.js:
```diff
@@ -967,11 +968,15 @@ export function save_message_row_edit($row) {
+                    compose_banner.CLASSNAMES.generic_compose_error,
+                    $container,
                 );
-                $row.find(".edit_error").text(message).show();
```
We don't know that the `xhr.responseText` is JSON -- it's an error response, and those are sometimes weird. See `ui_report.error` for some sample code that does the right thing (`xhr_error_message`, the function you're replacing, has similar logic):

```
export function error(                                                                                                                
    response_html: string,                                                                                                            
    xhr: JQuery.jqXHR | undefined,                                                                                                    
    status_box: JQuery,                                                                                                               
    remove_after?: number,                                                                                                            
): void {                                                                                                                             
    if (xhr && xhr.status >= 400 && xhr.status < 500) {                                                                               
        // Only display the error response for 4XX, where we've crafted                                                               
        // a nice response.                                                                                                           
        const server_response_html = _.escape(JSON.parse(xhr.responseText).msg);                                                      
        if (response_html) {                                                                                                          
            response_html += ": " + server_response_html;                                                                             
        } else {                                                                                                                      
            response_html = server_response_html;                                                                                     
        }                                                                                                                             
    }                                                                                                                                 
                                                                                                                                      
    message(response_html, status_box, "alert-error", remove_after);                                                                  
}                                                                                                                                     
                                                                                                                                      
```

shameondev replied:
I have made some tweaks to `xhr_error_message` to allow an optional message parameter, similar to how it's implemented in `ui_report.error`. It seems like we need a unified function to handle the task of returning `response_html`, but I am not sure if I should work on that right now and where to place it.

shameondev commented:
I have pushed some updates that will allow for the use of `xhr_error_message`. I believe it's now ready for the next review iteration.

timabbott commented on web/tests/channel.test.js:
```diff
@@ -343,6 +343,9 @@ test("xhr_error_message", () => {
     msg = "some message";
     assert.equal(channel.xhr_error_message(msg, xhr), "some message: file not found");
+
+    msg = "";
```
This test should use `null`, to match how it's used.

timabbott commented:
Merged after doing the tiny tweak in https://github.com/zulip/zulip/pull/25415/commits/5ff4f3ba6e2606f484367833086ca555c3d1996c#r1186485688, thanks @shameondev!
